src/Symfony/Component/Form/Exception/TransformationFailedException.php

```diff
@@ -18,11 +18,13 @@
  */
 class TransformationFailedException extends RuntimeException
 {
-    private ?string $invalidMessage;
-    private array $invalidMessageParameters;
-
-    public function __construct(string $message = '', int $code = 0, \Throwable $previous = null, string $invalidMessage = null, array $invalidMessageParameters = [])
-    {
+    public function __construct(
+        string $message = '',
+        int $code = 0,
+        \Throwable $previous = null,
+        private ?string $invalidMessage = null,
+        private array $invalidMessageParameters = [],
```

Comment:
I would revert the two lines above as the properties are actually initialised in `setInvalidMessage()`.

Reply:
cc @nicolas-grekas AFAIK you want to have a promoted property in those cases, right?